### PR TITLE
Generic media model

### PR DESCRIPTION
**Pull request by ziachap**

### CODE REVIEW

File: Gibe.DittoProcessors/Processors/ImagePickerAttribute.cs [hunk 1]
```diff
@@ -1,22 +1,17 @@
...
 {
-	public class ImagePickerAttribute : DittoProcessorAttribute
+	public class ImagePickerAttribute : InjectableProcessorAttribute
 	{
-		private readonly IMediaService _mediaService;
+		public Func<IMediaService> MediaService => Inject<IMediaService>();
+		private readonly Type _type;
 
-		public ImagePickerAttribute(IMediaService mediaService)
+		public ImagePickerAttribute(Type type = null)
```
Comment (stevetemple) [inline Gibe.DittoProcessors/Processors/ImagePickerAttribute.cs]: I'm guessing you can't have Type type = typeof(MediaImageModel)

File: Gibe.DittoProcessors/Processors/GridAttribute.cs [hunk 2]
```diff
@@ -44,7 +45,7 @@ public override object ProcessValue()
 						break;
 					case GridEditorAliases.Media:
 						var gridContentMediaValue = JsonConvert.DeserializeObject<GridContentMediaValue>(control.Value.ToString());
-						var mediaImage = _mediaService.GetImage(gridContentMediaValue.Id);
+						var mediaImage = _mediaService.GetImage<CaptionMediaImageModel>(gridContentMediaValue.Id);
```
Comment (stevetemple) [inline Gibe.DittoProcessors/Processors/GridAttribute.cs]: Is this the default for Grid that it'll be a CaptionMediaImageModel or is that Bristol Sport specific

File: Gibe.DittoProcessors/Processors/ImagePickerAttribute.cs (hunk 1 above)
Comment (ziachap) [inline Gibe.DittoProcessors/Processors/ImagePickerAttribute.cs]: yeah, can't have that.

File: Gibe.DittoProcessors/packages.config [hunk 3]
```diff
@@ -39,6 +39,7 @@
   <package id="Owin" version="1.0" targetFramework="net451" />
   <package id="semver" version="1.1.2" targetFramework="net451" />
   <package id="SharpZipLib" version="0.86.0" targetFramework="net451" />
+  <package id="System.ValueTuple" version="4.3.0" targetFramework="net452" />
```
Comment (stevetemple) [inline Gibe.DittoProcessors/packages.config]: Where does this get used?

Comment (ziachap) [inline Gibe.DittoProcessors/packages.config]: MediaService line 102, used to couple a media item together with whether it is a folder or not

File: Gibe.DittoProcessors/Processors/GridAttribute.cs (hunk 2 above)
Comment (ziachap) [inline Gibe.DittoProcessors/Processors/GridAttribute.cs]: The grid editor allows you to add a caption to each image so this needs to be in there

**Pull request merged**